clients/python-wrapper/lakefs/cmd.py

```diff
@@ -0,0 +1,221 @@
+    with open(file_path, 'rb') as f:
+        if platform_info.system == 'windows':
+            # on windows, Python places an executable EXE file.
+            # it contains the python snippet at the end of the file
+            f.seek(-4096, io.SEEK_END)  # last 4k
+            return b'__main__.py' in f.read(4096)
+        # on unix, the lakeFS binary is a Python script
+        first_line = f.readline()
+        return first_line.startswith(b'#!') and b'python' in first_line.lower()
+
+
+def _find_binary(binary_name: str) -> Optional[str]:
+    '''
+    Find the binary in PATH or ~/.lakefs/bin, skipping Python scripts.
+    Returns the path to the binary or None if not found.
+    '''
+    home_bin_path = os.path.expanduser(f'{BINARY_DOWNLOAD_DIR}/{binary_name}')
+    # Check if the binary is in the PATH
+    binary_path = shutil.which(binary_name)
+    # If the found binary is a Python script, find the next occurrence
+    platform_info =_get_platform_info()
+    if platform_info.system == 'windows':
+        binary_name = f'{binary_name}.exe'
+        home_bin_path = os.path.expanduser(f'{BINARY_DOWNLOAD_DIR}/{binary_name}')
+    if binary_path and _is_python_script(binary_path):
+        # Find the next occurrence of the binary in PATH
+        for path in os.environ.get('PATH', '').split(os.pathsep):
+            candidate = os.path.join(path, binary_name)
+            if os.path.isfile(candidate) and not _is_python_script(candidate):
+                binary_path = candidate
+                break
+        else:
+            binary_path = None
+    # If not found in PATH, check the home directory path
+    if not binary_path and os.path.isfile(home_bin_path):
+        binary_path = home_bin_path
+    return binary_path
+
+
+def _get_latest_version() -> str:
+    '''
+    Get the latest lakeFS release version from GitHub API
+    '''
+    try:
+        with urllib.request.urlopen(BINARY_LATEST_RELEASE_URL) as response:
+            data = json.loads(response.read().decode())
+            return data['tag_name'].lstrip('v')  # Remove 'v' prefix from version
+    except (urllib.error.URLError, json.JSONDecodeError, KeyError) as e:
+        raise RuntimeError(f"Error getting latest lakeFS version: {e}") from e
+
+
+class FakeProgressBar:
+    """A no-op progress bar used when tqdm is not available."""
+    def __enter__(self):
+        return self
+
+    def __exit__(self, _exc_type, _exc_val, _exc_tb):
+        pass
+
+    def update(self, _size: int):
+        """Update the progress bar with the given size (no-op)."""
+
+
+def _progress(total_size: int):
+    if tqdm:
+        return tqdm.tqdm(total=total_size, unit='iB', unit_scale=True)
+    return FakeProgressBar()
+
+
+def _download_file(url: str) -> io.BytesIO:
+    '''
+    Download a file from the given URL and return it as a BytesIO object
+    '''
+    content = io.BytesIO()
+    try:
+        req = urllib.request.Request(url, headers={'User-Agent': 'lakeFS SDK Downloader'})
```

Comment:
add the sdk version to the useragent

Reply:
I can however do keep in mind this is being sent to GitHub so I don't even know if we have access to logs

Reply:
not blocking becaue of that - we currently generate the user-agent string for the generated sdk. when we will merge both of them it will be easy to just the same user-anget to idetnify the client.